openfe_gromacs/protocols/gromacs_md/md_methods.py

```diff
@@ -654,3 +848,228 @@ def _execute(
+            mdp = [
+                x
+                for x in mdp_files
+                if str(x).split("/")[-1] == output_settings_em.mdp_file
+            ]
+            tpr = pathlib.Path(ctx.shared / output_settings_em.tpr_file)
+            assert len(mdp) == 1
+            # ToDo: If no traj should be written out, don't write empty file?
+            self._run_gromacs(
+                mdp[0],
+                input_gro,
+                input_top,
+                tpr,
+                output_settings_em.gro_file,
+                output_settings_em.xtc_file,
+                output_settings_em.trr_file,
+                output_settings_em.cpt_file,
+                output_settings_em.log_file,
+                output_settings_em.edr_file,
+                ctx.shared,
+            )
+
+        # ToDo: Should we disallow running MD without EM?
+        # Run NVT
+        if sim_settings_nvt.nsteps > 0:
+            if verbose:
+                self.logger.info("Running an NVT MD simulation")
+            mdp = [
+                x
+                for x in mdp_files
+                if str(x).split("/")[-1] == output_settings_nvt.mdp_file
+            ]
+            tpr = pathlib.Path(ctx.shared / output_settings_nvt.tpr_file)
+            assert len(mdp) == 1
+            # ToDo: Change .gro to output from EM if we do EM first,
+            # else original .gro file
+            self._run_gromacs(
+                mdp[0],
+                input_gro,
+                input_top,
+                tpr,
+                output_settings_nvt.gro_file,
+                output_settings_nvt.xtc_file,
+                output_settings_nvt.trr_file,
+                output_settings_nvt.cpt_file,
+                output_settings_nvt.log_file,
+                output_settings_nvt.edr_file,
+                ctx.shared,
+            )
+
+        # Run NPT MD simulation
+        if sim_settings_npt.nsteps > 0:
+            if verbose:
+                self.logger.info("Running an NPT MD simulation")
+            mdp = [
+                x
+                for x in mdp_files
+                if str(x).split("/")[-1] == output_settings_npt.mdp_file
+            ]
+            tpr = pathlib.Path(ctx.shared / output_settings_npt.tpr_file)
+            assert len(mdp) == 1
+            self._run_gromacs(
+                mdp[0],
+                input_gro,
+                input_top,
+                tpr,
+                output_settings_npt.gro_file,
+                output_settings_npt.xtc_file,
+                output_settings_npt.trr_file,
+                output_settings_npt.cpt_file,
+                output_settings_npt.log_file,
+                output_settings_npt.edr_file,
+                ctx.shared,
+            )
+
+        return {
+            "repeat_id": self._inputs["repeat_id"],
+            "generation": self._inputs["generation"],
+            "gro_em": shared_basepath / output_settings_em.gro_file,
```

Comment:
A lot of these aren't in the results unit

Reply:
Added these.